module/VuFind/tests/unit-tests/src/VuFindTest/Db/Service/SessionServiceTest.php

```diff
@@ -0,0 +1,319 @@
+            ->with($this->anything())
+            ->willReturn($session);
+        $session->expects($this->once())->method('setData')
+            ->with('foo')
+            ->willReturn($session);
+        $service = $this->getService($entityManager, $pluginManager);
+        $this->assertEquals(true, $service->WriteSession('1', 'foo'));
+    }
+
+    /**
+     * Test destroying the session.
+     *
+     * @return void
+     */
+    public function testDestroySession()
+    {
+        $session = $this->getMockBuilder(\VuFind\Db\Entity\Session::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $entityManager = $this->getEntityManager($session);
+        $pluginManager = $this->getPluginManager(true);
+        $queryBuilder = $this->getMockBuilder(\Doctrine\ORM\QueryBuilder::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $queryBuilder->expects($this->once())->method('delete')
+            ->with(Session::class, 's')
+            ->willReturn($queryBuilder);
+        $queryBuilder->expects($this->once())->method('where')
+            ->with('s.sessionId = :sid')
+            ->willReturn($queryBuilder);
+        $queryBuilder->expects($this->once())->method('setParameter')
+            ->with('sid', 1)
+            ->willReturn($queryBuilder);
+        $query = $this->getMockBuilder(\Doctrine\ORM\AbstractQuery::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['execute'])
+            ->getMockForAbstractClass();
+        $query->expects($this->once())->method('execute')
+            ->willReturn($this->anything());
+        $queryBuilder->expects($this->once())->method('getQuery')
+            ->willReturn($query);
+        $entityManager->expects($this->once())->method('createQueryBuilder')
+            ->willReturn($queryBuilder);
+        $service = $this->getService($entityManager, $pluginManager);
+        $service->destroySession('1');
+    }
+
+    /**
+     * Test destroying the expired sessions.
+     *
+     * @return void
+     */
+    public function testGarbageCollect()
+    {
+        $session = $this->getMockBuilder(\VuFind\Db\Entity\Session::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $entityManager = $this->getEntityManager($session);
+        $pluginManager = $this->getPluginManager(true);
+        $queryBuilder = $this->getMockBuilder(\Doctrine\ORM\QueryBuilder::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $queryBuilder->expects($this->once())->method('delete')
+            ->with(Session::class, 's')
+            ->willReturn($queryBuilder);
+        $queryBuilder->expects($this->once())->method('where')
+            ->with('s.lastUsed < used')
+            ->willReturn($queryBuilder);
+        $queryBuilder->expects($this->once())->method('setParameter')
+            ->with('used', time() - 10000)
```

Comment:
I imagine there's a chance (probably quite small, but not impossible) that this test could occasionally fail due to a timing issue, if the second rolls over between the execution of this line of code, and the running of the rest of the code. It might be a good idea to replace the `time() - 10000` expectation here with something a little looser. I believe you can use `$this->equalTo(time() - 10000, 1)` to give it a delta value of 1 to avoid the problem.